tests/test_azure_blob.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import array
+import hashlib
+import logging
+import os
+import random
+import re
+import unittest
+import uuid
+from contextlib import contextmanager
+from tempfile import TemporaryDirectory
+from typing import NamedTuple
+
+from iopath.common.azure_blob import AzureBlobPathHandler, AzureBlobTokenProvider
+
+
+try:
+    import azure.storage.blob as azure_blob
+except ImportError:
+    azure_blob = None
+
+
+logging.basicConfig(
+    format="%(asctime)s %(message)s", datefmt="%m/%d/%Y %I:%M:%S %p", level=logging.INFO
+)
+
+ENV_SAS_TOKEN = "AZURE_STORAGE_SAS_TOKEN"
+ENV_BASE_PATH = "AZURE_TEST_BASE_PATH"
+
+
+class EnvironmentTokenProvider(AzureBlobTokenProvider):
+    def get_sas_token(self, _: str) -> str:
+        assert (
+            ENV_SAS_TOKEN in os.environ
```

Comment:
How is this token generated? Maybe you should add a reference to to the Azure doc that explains this token generation.

Reply:
Also, what are other ways user can provide a token? Maybe also make passing of the token provider optional so that the Azure Path handler can figure it out.

Reply:
I would expect environment variables to be the easiest and most common way, but I could imagine other approaches that integrate with a separate secrets store or key vault.

You can also use AAD token credentials as described here: https://learn.microsoft.com/en-us/python/api/overview/azure/storage-blob-readme?view=azure-python

But that's a different mechanism which I haven't seen used internally yet so I didn't add anything for it yet.

I'll update it to fall back to the env provider if one isn't specified.